### ISSUE TITLE
garbage-collection warning despite using shutdown -- only when dbdir is specified

### ISSUE THREAD

**Issue by cboettig** [duckdb/duckdb-r#60]
When the connection is created with disk-base storage, the following gives the infamous garbage-collection warning (#34, #58) when `gc()` is called, despite already being correctly shutdown as directed:

```r
dir <- tempfile()
con <- DBI::dbConnect(duckdb::duckdb(), dir)
DBI::dbDisconnect(con, shutdown=TRUE)
gc()

#          used (Mb) gc trigger (Mb) max used (Mb)
# Ncells  732512 39.2    1403478   75  1403478 75.0
# Vcells 1236192  9.5    8388608   64  2328637 17.8
# Warning message:
# Database is garbage-collected, use dbDisconnect(con, shutdown=TRUE) or duckdb::duckdb_shutdown(drv) to avoid this. 
```

Without the dbdir argument, we do not get a warning:

```r
con <- DBI::dbConnect(duckdb::duckdb())
DBI::dbDisconnect(con, shutdown=TRUE)
gc()
```

Any advice on suppressing this warning?  (I have a few packages that wrap duckdb and users are frequently confused by seeing this warning, mistaking it for an error).

**Comment by kent37** [comment]
I'm seeing the same issue when using `withr::defer()` to call `dbDisconnect()`. This may be associated with upgrading  duckdb to 0.9.2-1 from 0.8.1.

**Comment by krlmlr** [comment]
It's likely related to #83.

**Comment by kent37** [comment]
A workaround is to pass `dir` (and other parameters such as `read_only`) to `duckdb()` directly.

```
dir <- tempfile()
con <- DBI::dbConnect(duckdb::duckdb(dbdir=dir))
DBI::dbDisconnect(con, shutdown=TRUE)
gc()
          used  (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)
Ncells 2781362 148.6    4843155 258.7         NA  4843155 258.7
Vcells 6018443  46.0   12255594  93.6      16384  9301840  71.0
```

**Comment by krlmlr** [comment]
This looks much better in #124, I'll merge it today, binaries will be available on https://duckdb.r-universe.dev/duckdb# soon. Can you confirm?

**Comment by krlmlr** [comment]
Is this still an issue with v0.10.1, on CRAN now?

**Comment by kent37** [comment]
Yes, it is fixed for me in v0.10.1. Neither the reprex at the start of this issue nor my presenting issue gives the "Database is garbage-collected" warning.
